pandas/core/dtypes/inference.py

```diff
@@ -282,12 +285,20 @@ def is_list_like(obj):
     >>> is_list_like(np.array(2)))
     False
     """
-
-    return (isinstance(obj, compat.Iterable) and
-            # we do not count strings/unicode/bytes as list-like
-            not isinstance(obj, string_and_binary_types) and
-            # exclude zero-dimensional numpy arrays, effectively scalars
-            not (isinstance(obj, np.ndarray) and obj.ndim == 0))
+    if strict is None and isinstance(obj, set):
+        # only raise warning if necessary
+        warnings.warn('is_list_like will in the future return False for sets. '
```

Comment:
Is there a consensus that we want this to be the default future behavior instead of non-default option? I'm not convinced.  Would be interesting to see how many of the existing uses of `is_list_like` are valid vs. invalid for sets.

Reply:
this whole idea was just that - just an idea
i don’t think we actually want to add a kwarg at all

the idea was to change it within pandas only (if this is useful)
and not actually change the external api at all

Reply:
Since `list` is ordered in python, the name `is_list_like` suggests the same property. If that's not the case it's good ("explicit is better than implicit") to have to specify it.

Don't see why that shouldn't apply to external API as well (with deprecation cycle etc.)